assembly/src/assembler/context.rs

```diff
@@ -428,7 +428,7 @@ impl ModuleContext {
         if self.compiled_procs.iter().any(|p| p.name() == name)
             || self.proc_stack.iter().any(|p| &p.name == name)
         {
-            return Err(AssemblyError::duplicate_proc_name(name, &self.path));
+            event!(Level::WARN, "duplicate proc name '{}' in module {}", name, &self.path);
```

Comment:
What would happen if we have two notes with the same procedure name, but different implementations? If I understood correctly, this would favor the first implementation, and use the same code for both notes. If that is the case, I think we need some different approach, perhaps compile both, and check that the MASTs are the same?

Reply:
Shouldn't this become a non-issue when we switch to using MAST as the compilation artifact? In other words, notes will be compiled separately to MAST ahead-of-time, rather than from MASM on-demand. Then a transaction script program (also in MAST form), would either reference the notes by MAST root, or would inline the MAST for the note into itself. To be clear, I know that isn't the case right now, but I'm referring to once we make the changes described in #1226 and #1217 and switch away from using MASM as the artifact.

Reply:
See also 0xPolygonMiden/compiler#132, which is a draft spec for those changes as well as the binary format for the MAST. The draft is being worked on there temporarily, it will be moved to this repo once I've worked through the first iteration of feedback.

Reply:
I think it should solve this issue indeed!

Reply:
Should we punt on solving this problem temporarily until that's done then? I think we're basically at the point where we can start doing the work to make that happen, we've just been taking our time with working through some of the details - but the PR I linked is basically the gathering the results of all those discussions together into a draft spec, and once we're happy with it, we can start implementing basically right away. 

Bobbin recently mentioned in one of our conversations that you would probably be doing the implementation @Overcastan, though that's obviously not set in stone, but if you have any thoughts on the draft spec before I open it as a proper proposal here in the `miden-vm` repo, feel free to jump in as a reviewer.

In the meantime, it seems like it might be best to make the case of duplicate procedure names with differing MAST roots an error, to avoid accidentally treating them as equivalent. Then, once we implement the MAST changes, the practical issue with note scripts goes away.

Reply:
Yes - I think it makes sense to hold off on solving this as it will become a non-issue once we use MAST-based format to represent note scripts and account code. We should start working more actively on this starting next week, and hopefully, within 2 - 3 weeks we can be in a place where is issue becomes irrelevant.

Reply:
I think this issue is basically in the same boat as #1211.